### ISSUE TITLE
Make dump include a newline at the end, similar to write step

### ISSUE THREAD

**Issue by siowyisheng** [kelaberetiv/TagUI#654]
Proposal: For `dump` and `write`, include a newline after the text.

Rationale: It makes repeated `writes` to the same file behave in the more commonly desired way, and avoids requiring business users to learn `\n`.

**Comment by kensoh** [comment]
Actually write step does that already, in tagui_header.js, and it uses `\r\n` -

```js
// for appending text information to file
function append_text(file_name,info_text) {var ds; if (flow_path.indexOf('/') !== -1) ds = '/'; else ds = '\\';
if (!file_name) {if (save_text_count==0) save_text_count++; // increment if 0, else use same count to append
file_name = flow_path + ds + 'text' + save_text_count.toString() + '.txt';}
var fs = require('fs'); fs.write(file_name, info_text + '\r\n', 'a');}
```

But dump step below does not. Am on the same page having users putting in `\r\n` is too inconvenient, so I include that as part of write step, but didn't include in dump step in order to preserve the source text fully.

```js
// for saving text information to file
function save_text(file_name,info_text) {var ds; if (flow_path.indexOf('/') !== -1) ds = '/'; else ds = '\\';
if (!file_name) {save_text_count++; file_name = flow_path + ds + 'text' + save_text_count.toString() + '.txt';}
var fs = require('fs'); fs.write(file_name, info_text, 'w');}
```

The trade-off of including in write step will be there is no way from using TagUI steps to write to a file without appending new lines. Ie can't write consecutive strings on the same line. But I think that is like an super edge case for users to want to write stuffs without new lines.

For dump step, if change to include `\r\n` it will be compatibility-breaking, but it is good to do that now in v6. Why breaking, cuz people use dump and with newline characters to create the csv header files, then use write to append lines. Changing this will lead to empty line after header.

Of course, can do some smart detection that if string already ends with newlines then don't add on. But that's a catch in that the behaviour is not consistent. Eg what if I want to dump line 1 normal line and a line 2 blank line? Eg up becomes a line only.

**Comment by siowyisheng** [comment]
I see. My bad. You're right it's only `dump` that doesn't do that. I think `dump` should do it anyway.

**Comment by kensoh** [comment]
I think so too, should definitely include this in v6, removing discussion label. Then both dump and write will be consistent, some users will have to make some adjustments. But this would make the tool and language better.

**Comment by kensoh** [comment]
raised PR to merge change into dev branch, unassigning myself

**Comment by siowyisheng** [comment]
implemented in #673 for v6
